packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -644,7 +646,7 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with AutomaticK
           rendererIgnoresPointer: true,
           cursorWidth: widget.cursorWidth,
           cursorRadius: widget.cursorRadius,
-          cursorColor: widget.cursorColor,
+          cursorColor: widget.cursorColor ?? CupertinoTheme.of(context).primaryColor,
```

Comment:
small detail but since themeData is already pulled out above, you can just write `themeData.primaryColor` here.